message.drush.inc

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/**
+ * @file
+ * Defines drush commands.
+ */
+
+
+/**
+ * Implements hook_drush_command().
+ */
+function message_drush_command() {
+  $items = array();
+  $items['message-purge'] = array(
+    'description' => t('Batch purge old messages (will only affect messages that are eligible for purging based on message settings).'),
+    'arguments' => array(
+      'purge_limit' => 'The total number of messages to purge. Defaults to 10,000. Set to -1 to purge all eligible messages (warning: this can cause memory limit errors for very large sets).',
+      'loop_limit' => 'The number of messages to purge per loop. Defaults to 100.',
+    ),
+    'bootstrap' => DRUSH_BOOTSTRAP_DRUPAL_FULL,
+    'aliases' => array('purge-messages'),
+    'drupal dependencies' => array('message'),
+    'callback' => 'drush_message_purge',
+  );
+  return $items;
+}
+
+/**
+ * Drush command to batch purge old messages.
+ */
+function drush_message_purge() {
+  drush_print("Starting message purge.");
+
+  // Get drush params.
+  $args = drush_get_arguments();
+  $purge_limit = MESSAGE_PURGE_LIMIT_DRUSH_DEFAULT;
+  $loop_limit = MESSAGE_PURGE_LOOP_DRUSH_DEFAULT;
+  if (isset($args[1]) && is_numeric($args[1])) {
+    $purge_limit = $args[1];
+  }
+  if (isset($args[2]) && is_numeric($args[2])) {
+    $loop_limit = $args[2];
+  }
+  if ($purge_limit > 0) {
+    drush_print("Purge Limit: $purge_limit");
+  }
+  else {
+    drush_print("Purge Limit: no limit (purge all eligible messages)");
+  }
+  drush_print("Loop Limit: $loop_limit");
+
+  $completed = 0;
+  do {
```

Comment:
I don't like the do while loop. Implement a batch operation when deleting a 500 everey time when this would be a drush arguments that can be changed when passing in the command.